src/main/java/com/google/sps/servlets/utils/CommentAnalysis.java

```diff
@@ -53,31 +53,56 @@ public CommentAnalysis(LanguageServiceClient languageService) {
    * @return a Statistics object that contains required values to display
    */
   public Statistics computeOverallStats(CommentThreadListResponse youtubeResponse) {
-    List<Double> scoreValues =
+    // all sentiments built by comment content retrieved from youtubeResponse
+    List<Sentiment> documentList =
         youtubeResponse.getItems().stream()
             .map(UserComment::new)
-            .map(this::calcualateSentiAnalysisScore)
+            .map(
+                comment ->
+                    languageService
+                        .analyzeSentiment(
+                            Document.newBuilder()
+                                .setContent(comment.getCommentMsg())
+                                .setType(Document.Type.PLAIN_TEXT)
+                                .build())
+                        .getDocumentSentiment())
+            .collect(Collectors.toList());
+    // List of scores and magnitude for all comments
+    List<Double> scoreValues =
+        documentList.stream().map(this::calcualateSentiAnalysisScore).collect(Collectors.toList());
+    List<Double> magnitudeValues =
+        documentList.stream()
+            .map(this::calculateSentiAnalysisMagnitude)
             .collect(Collectors.toList());
-    return new Statistics(scoreValues);
+    return new Statistics(scoreValues, magnitudeValues);
   }
 
   /**
    * Perform sentiment analysis of comment.
    *
+   * @param sentiment object retrieved from document for each comment
    * @return sentiment score
    * @throws RuntimeException if the sentiment analysis API is not working, throw the IOExeption
    */
-  private double calcualateSentiAnalysisScore(UserComment comment) {
-    Document doc =
-        Document.newBuilder()
-            .setContent(comment.getCommentMsg())
-            .setType(Document.Type.PLAIN_TEXT)
-            .build();
-    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+  private double calcualateSentiAnalysisScore(Sentiment sentiment) {
```

Comment:
The method name doesn't fit the content anymore. You are not calculateSentimentAnalysis anymore, but extracting the score from the Sentiment. Find a better name? And the method below as well.
Please also update the comments for the method.

Also, do you still need the RuntimeException? It says it is for when the API is not working, but you have already retrieved the Sentiment above. Maybe move the exception to above when you call languageService.analyzeSentiment? It may be worthwhile to retrieve Sentiment in its own separate method.